File: solutions/3227-find-missing-and-repeated-values/solution.py
class Solution:
    def findMissingAndRepeatedValues(self, grid: List[List[int]]) -> List[int]:
        seen = set()
        n = len(grid)

        for row in grid:
            for el in row:
                if el in seen:
                    a = el
                else:
                    seen.add(el)

        for i in range(1, n*n + 1):
            if i not in seen:
                return [a, i]

File: solutions/3227-find-missing-and-repeated-values/solution.js
/**
 * @param {number[][]} grid
 * @return {number[]}
 */
var findMissingAndRepeatedValues = function(grid) {
    const seen = new Set()
    let a, b = null

    for(const row of grid){
        for(const el of row){
            if(!seen.has(el)){
                seen.add(el)
            }else{
                a = el
            }
        }
    }

    for(let i = 1; i <= grid.length ** 2; i++){
        if(!seen.has(i)){
            b = i
            return [a,b]
        }
    }
};
